### PR TITLE
PR - Mail lEvent 

### PR DESCRIPTION
**Pull request by GersonVallejos**

### CODE REVIEW

File: src/main/java/com/backend/taskagile/controller/UserController.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import lombok.AllArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
```
Comment (cvallejosh1) [inline src/main/java/com/backend/taskagile/controller/UserController.java:9]: remove unused import

File: src/main/java/com/backend/taskagile/event/EmailEvent.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.backend.taskagile.event;
+
+import com.backend.taskagile.model.User;
+import org.springframework.context.ApplicationEvent;
+
+public class EmailEvent extends ApplicationEvent {
```
Comment (cvallejosh1) [inline src/main/java/com/backend/taskagile/event/EmailEvent.java:6]: It doesn't need extend ApplicationEvent

File: src/main/java/com/backend/taskagile/service/UserService.java [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.stereotype.Service;
 
 @Service
 @AllArgsConstructor
 public class UserService {
 
+  private ApplicationEventPublisher applicationEventPublisher;
   private final UserRepository userRepository;
+  //private final EmailSender emailSender;
```
Comment (cvallejosh1) [inline src/main/java/com/backend/taskagile/service/UserService.java:19]: remove unused.

File: src/main/java/com/backend/taskagile/event/EmailEventListener.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  private final EmailSender emailSender;
+
+  @EventListener
+  public void handleEvent(EmailEvent event) {
+    emailSender.send(event.getUser().getEmail(),
+        buildEmail(event.getUser().getFirstName()));
+
+  }
+
+  private String buildEmail(String name) {
```
Comment (cvallejosh1) [inline src/main/java/com/backend/taskagile/event/EmailEventListener.java:21]: Use templates to build the email body

**Review by cvallejosh1 — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -6,3 +6,14 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
 spring.jpa.hibernate.ddl-auto=create-drop
 
+spring.mail.host=localhost
```
Comment (cvallejosh1) [inline src/main/resources/application.properties:9]: It is required to install something else on your local to set up the mail server?

**Review by cvallejosh1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/backend/taskagile/controller/UserController.java (hunk 1 above)
Comment (GersonVallejos) [reply src/main/java/com/backend/taskagile/controller/UserController.java:9]: removed

File: src/main/java/com/backend/taskagile/event/EmailEvent.java (hunk 2 above)
Comment (GersonVallejos) [reply src/main/java/com/backend/taskagile/event/EmailEvent.java:6]: Removed

File: src/main/java/com/backend/taskagile/service/UserService.java (hunk 3 above)
Comment (GersonVallejos) [reply src/main/java/com/backend/taskagile/service/UserService.java:19]: removed

File: src/main/resources/application.properties (hunk 5 above)
Comment (GersonVallejos) [reply src/main/resources/application.properties:9]: It is required to use "npm install -g maildev"

**Pull request merged**